src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/CustomConverterTests/CustomConverterTests.cs

```diff
@@ -181,5 +182,35 @@ public static void GetConverterTypeToConvertNull()
         {
             Assert.Throws<ArgumentNullException>(() => (new JsonSerializerOptions()).GetConverter(typeToConvert: null!));
         }
+
+        [Fact]
+        public static void CustomStringConverter_UsedInDictionaries()
+        {
+            var options = new JsonSerializerOptions
+            {
+                Converters = { new CustomStringConverter() },
+                DictionaryKeyPolicy = JsonNamingPolicy.CamelCase
```

Comment:
Out of curiosity, why are you adding a key policy into the mix?

Reply:
We should be testing that the serializer honors the setting and it also helps verify that the custom converter is not being called here. I'll add a variant that doesn't include the policy.